integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/adapters.py

```diff
@@ -98,6 +98,10 @@ class AnthropicClaudeAdapter(BedrockModelAdapter):
     Adapter for the Anthropic Claude models.
     """
 
+    def __init__(self, model_kwargs: Dict[str, Any], max_length: Optional[int]) -> None:
+        self.use_messages_api = model_kwargs.get("use_messages_api", True)
```

Comment:
I know this is how we handled it in v1, but I wonder if exposing `use_messages_api` as an actual param instead of inside `model_kwargs` would be a good idea so we could add a docstring for it.

Reply:
For this we would need to change how Adapters are initialized in general. See https://github.com/deepset-ai/haystack-core-integrations/blob/cbb51ed2c0613ee2e42a1eee303283b4d00ecb78/integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/generator.py#L141-L142

Even if we did, the impact would be quite limited as usually a user doesn't interact with the adapter. 
We could add a docstring to `AmazonBedrockGenerator`'s init instead. But so far we haven't added any model-specific params there. As this is more there for legacy reasons, I wouldn't bother for now... WDYT?

Reply:
All good points! I'm fine to leave as is.